### ISSUE TITLE
configuring logrotate

### ISSUE THREAD

**Issue by gagou9** [cubing/worldcubeassociation.org#44]
As we handle the OS, we have do handle the logs as well or it's all gonna get messy :)

**Comment by jfly** [comment]
Removing URGENT label. I've moved us to a server with 16GB of space, and we're currently using less than half. We should be ok for a while yet. That said, this is still something important to do. @gagou9, would you be up for taking this task on?

**Comment by gagou9** [comment]
> @gagou9, would you be up for taking this task on?
For sure.

**Comment by jbcm627** [comment]
Any progress on this? The nginx and Ruby logs have already grown to ~1 GB
each.

**Comment by jfly** [comment]
Bump. Any news, Gael?

**Comment by gagou9** [comment]
Yo,

I just gave a look and manually rotated the logs as explained in nginx doc.
The access.log.0.gz was 1.3gb and is now 80mb.

Using the nginx package from ubuntu would ship the logrotate config, unfortunately we don't do that at the moment, so I bet I'll have to dig into chef recipes...

Here's what I found (I still don't have a testing environment so couldn't actually try it) :
https://supermarket.chef.io/cookbooks/logrotate

I made that recipe, but as I said couldn't try it.
The line I'm not
```chef
logrotate_app 'nginx' do
  cookbook  'logrotate'
  path      ['/var/log/nginx/access.log', '/var/log/nginx/error.log']
  frequency 'daily'
  rotate    52
  create    '644 www-data root'
  options  ['missingok', 'compress', 'delaycompress', 'notifempty', 'sharedscripts']
  'postrotate' '[ -f /var/run/nginx.pid ] && kill -USR1 `cat /var/run/nginx.pid`'
end
```

**Comment by jfly** [comment]
Yeah, for better or worse, we're building nginx from source. Hopefully adding a chef recipe shouldn't be too hard. The postrotate line does look weird, not sure that that's valid ruby/that it does what you want...

I do not plan on working on this anytime soon, so Gael, if you have time, you should feel free to dig in. Your highest priority should be getting your nginx changes into git, though. Thanks!

**Comment by gagou9** [comment]
Hey, I just edited my previous command, didn't see an obvious mistake (the quotes arround postrotate). The rest seems fine, doesn't it ?

I have made the nginx changes on my local git repo, will push soon. But I need to find a good way to have a test environment, the problem beeing that I'm on about 10 different geek projects, thus I don't want my laptop to get messy, so installing stuff and stuff and stuff doesn't make me comfortable!

And I won't have much time in the next weeks, but will try my best. thanks!
